Kwf/Media/Image.php

```diff
@@ -550,8 +550,34 @@ public static function scale($source, $size, $uploadId = null, $sourceSize = nul
             $scaler = Kwf_Media_Scaler_Abstract::getInstance();
         }
 
+        if ($source instanceof Imagick) {
+            $im = $source;
+        } else {
+            $blob = file_get_contents($source);
+            Kwf_Util_Upload::onFileRead($source);
+            if (!strlen($blob)) throw new Kwf_Exception("File is empty");
+            $im = self::_createImagickFromBlob($blob, $mimeType);
+        }
+
+        if (method_exists($im, 'setImageChannelDepth') && $im->getImageChannelDepth(Imagick::CHANNEL_ALL) >= 16) {
+            $skipCleanup = false;
+        } else {
+            $skipCleanup = $preScale['factor'] > 0;  //preScale does this already
+        }
```

Comment:
Instead of Line 562-566 I suggest:
```
$skipCleanup = $preScale['factor'] > 0;
if ($skipCleanup) {
    $skipCleanup = $scaler->isSkipCleanup($source);
}
```

The method isSkipCleanup($source) should be implemented in every Scaler. In Kwf_Media_Scaler_Imagick you can check for `method_exists($im, 'setImageChannelDepth') && $im->getImageChannelDepth(Imagick::CHANNEL_ALL) >= 16` (and reuse the helper functions which you have duplicated here). In the other scalers you can skip this check (for now).

Reply:
done